docs/pages/web/daterange.js

```diff
@@ -0,0 +1,333 @@
+  'zh-CN': { localeData: zhCN, lang: 'Chinese (Simplified)' },
+  'zh-TW': { localeData: zhTW, lang: 'Chinese (Traditional)' },
+};
+
+export default function DatePickerPage({ generatedDocGen }: {| generatedDocGen: DocGen |}): Node {
+  const [locale, setLocale] = useState<string | null>(null);
+
+  return (
+    <Page title={generatedDocGen?.displayName}>
+      <PageHeader
+        badge="pilot"
+        name={generatedDocGen?.displayName}
+        description={generatedDocGen?.description}
+        slimBanner={
+          <SlimBanner
+            type="warning"
+            iconAccessibilityLabel="Warning message"
+            message="DateRange is an pilot component. Expect development and design iteration and breaking API changes."
+          />
+        }
+      >
+        <SandpackExample
+          code={main}
+          name={`Main ${generatedDocGen?.displayName} example`}
+          hideEditor
+          previewHeight={500}
+        />
+      </PageHeader>
+
+      <GeneratedPropTable generatedDocGen={generatedDocGen} />
+
+      <MainSection name="Usage guidelines">
+        <MainSection.Subsection columns={2}>
+          <MainSection.Card
+            type="do"
+            title="When to use"
+            description={`
+- When asking users to select past, present or future date ranges
+- When users can pick between pre-selected date ranges or input a custom date range
+`}
+          />
+          <MainSection.Card
+            type="don't"
+            title="When not to use"
+            description={`
+- When users need to select one specific day. Use [DatePicker](/web/datepicker) instead
+- When users need to input a date value with a numeric keyboard, for example when adding a birthday date. Use [DateField](/datefield) instead
+`}
+          />
+        </MainSection.Subsection>
+      </MainSection>
+
+      <MainSection name="Best practices">
+        <MainSection.Subsection columns={2}>
+          <MainSection.Card
+            type="do"
+            title="Do"
+            description={`
+- Disable future or past dates according to the use case. If the user would like to see predictions, for example, disable the past.
+- When possible, provide a list of applicable date ranges to facilitate user selection
+`}
+          />
+          <MainSection.Card
+            type="don't"
+            title="Don't"
+            description={`
+- Enable users to select dates in the future or past, if those dates are not a valid input
+- Provide a long of a list of applicable date ranges with confusing labels, to avoid confusing the user. Display concise options and in a logical order.
+`}
+          />
+        </MainSection.Subsection>
+      </MainSection>
+
+      <MainSection name="Variants">
+        <MainSection.Subsection
+          title="Controlled component"
+          description={`
+DateRange is a controlled component. Follow the implementation in the example to implement a controlled DateRange correctly.
```

Comment:
nit: I think we should this[ exact same format](https://gestalt.pinterest.systems/web/datefield#Controlled-component), and list out the props

Reply:
DONE